### PR TITLE
Feanil/programs pre supervisor

### PR DESCRIPTION
**Pull request by feanil**
@edx/devops

### CODE REVIEW

File: playbooks/roles/supervisor/templates/etc/init/pre_supervisor.conf.j2 [hunk 1]
```diff
@@ -5,4 +5,10 @@ task
...
+{% if programs_code_dir is defined %}
```
Comment (feanil) [inline playbooks/roles/supervisor/templates/etc/init/pre_supervisor.conf.j2:8]: If we like this approach, I'm happy to put in a separate PR to pull out the rest of the services to their own blocks instead of a single un-readable line.

File: playbooks/roles/supervisor/files/pre_supervisor_checks.py [hunk 2]
```diff
@@ -156,8 +166,8 @@ def edp_for_instance(instance_id):
                 play=play,
                 instance_id=instance_id)
             break
-        except:
-            print("Failed to get EDP for {}".format(instance_id))
+        except Exception as e:
+            print("Failed to get EDP for {}: {}".format(instance_id, str(e)))
```
Comment (jibsheet) [inline playbooks/roles/supervisor/files/pre_supervisor_checks.py:170]: :fireworks:

File: playbooks/roles/supervisor/templates/etc/init/pre_supervisor.conf.j2 (hunk 1 above)
Comment (jibsheet) [inline playbooks/roles/supervisor/templates/etc/init/pre_supervisor.conf.j2:8]: I find this somewhat more readable and would review a later PR to change it.

**Comment by jibsheet** [comment]
:+1:
